lib/features/identity_creator/presentation/identity_creator_controller.dart

```diff
@@ -118,59 +142,71 @@ class IdentityCreatorController extends BaseController {
 
   @override
   void onDone() {
-    viewState.value.fold(
-        (failure) {
-          if (failure is GetAllIdentitiesFailure) {
-            _getALlIdentitiesFailure(failure);
-          }
-        },
-        (success) {
-          if (success is GetAllIdentitiesSuccess) {
-            _getALlIdentitiesSuccess(success);
-          }
+    viewState.value.fold((failure) {
+      if (failure is GetAllIdentitiesFailure) {
+        createNewIdentityViewState.value = Left(failure);
+        _getALlIdentitiesFailure(failure);
+      }
+    }, (success) {
+      if (success is GetAllIdentitiesSuccess) {
+        if (isInCreatingProcess) {
+          _getAllDefaultIdentities(success.identities);
+          _handleCreateNewIdentity();
+        } else {
+          _getALlIdentitiesSuccess(success);
         }
-    );
+      }
+    });
   }
 
   void _setUpValueFromIdentity() {
     _nameIdentity = identity?.name ?? '';
     inputNameIdentityController?.text = identity?.name ?? '';
 
     if (identity?.textSignature?.value.isNotEmpty == true) {
-      signaturePlainEditorController?.text = identity?.textSignature?.value ?? '';
+      signaturePlainEditorController?.text =
+          identity?.textSignature?.value ?? '';
     }
 
     if (identity?.htmlSignature?.value.isNotEmpty == true) {
       updateContentHtmlEditor(identity?.htmlSignature?.value ?? '');
-      signatureHtmlEditorController.setText(identity?.htmlSignature?.value ?? '');
+      signatureHtmlEditorController
+          .setText(identity?.htmlSignature?.value ?? '');
     }
   }
 
   void _getAllIdentities() {
     log('IdentityCreatorController::_getAllIdentities() ');
     if (accountId != null) {
-      consumeState(_getAllIdentitiesInteractor.execute(
-        accountId!,
-        properties: Properties({'email'})
-      ));
+      consumeState(_getAllIdentitiesInteractor.execute(accountId!,
+          properties: Properties({'email'})));
     }
   }
 
   void _getALlIdentitiesSuccess(GetAllIdentitiesSuccess success) {
+    createNewIdentityViewState.value = Right(success);
     if (success.identities?.isNotEmpty == true) {
-      listEmailAddressDefault.value = success.identities
-          !.map((identity) => identity.toEmailAddressNoName())
+      listEmailAddressDefault.value = success.identities!
+          .map((identity) => identity.toEmailAddressNoName())
           .toSet()
           .toList();
       listEmailAddressOfReplyTo.add(noneEmailAddress);
       listEmailAddressOfReplyTo.addAll(listEmailAddressDefault);
-
       _setUpAllFieldEmailAddress();
     } else {
       _setDefaultEmailAddressList();
     }
   }
 
+  void _getAllDefaultIdentities(List<Identity>? allIdentities) {
+    final smallestIdentities =
+        IdentityUtils.instance.getSmallestOrderedIdentity(allIdentities);
```

Comment:
```suggestion
    final smallestIdentities = IdentityUtils.instance
      .getSmallestOrderedIdentity(allIdentities);
```
or something like that